packages/pds/src/account-manager/oauth-store.ts

```diff
@@ -0,0 +1,485 @@
+import { Client, createOp as createPlcOp } from '@did-plc/lib'
+import { Selectable } from 'kysely'
+import { Keypair, Secp256k1Keypair } from '@atproto/crypto'
+import {
+  Account,
+  AccountInfo,
+  AccountStore,
+  AuthenticateAccountData,
+  Code,
+  DeviceAccountInfo,
+  DeviceData,
+  DeviceId,
+  DeviceStore,
+  FoundRequestResult,
+  HandleUnavailableError,
+  InvalidRequestError,
+  NewTokenData,
+  RefreshToken,
+  RequestData,
+  RequestId,
+  RequestStore,
+  ResetPasswordConfirmData,
+  ResetPasswordRequestData,
+  SignUpData,
+  TokenData,
+  TokenId,
+  TokenInfo,
+  TokenStore,
+  UpdateRequestData,
+} from '@atproto/oauth-provider'
+import {
+  AuthRequiredError as XrpcAuthRequiredError,
+  InvalidRequestError as XrpcInvalidRequestError,
+} from '@atproto/xrpc-server'
+import { ActorStore } from '../actor-store/actor-store'
+import { BackgroundQueue } from '../background'
+import { ImageUrlBuilder } from '../image/image-url-builder'
+import { ServerMailer } from '../mailer'
+import { Sequencer } from '../sequencer'
+import { AccountManager } from './account-manager'
+import { AccountStatus, ActorAccount } from './helpers/account'
+import * as authRequest from './helpers/authorization-request'
+import * as device from './helpers/device'
+import * as deviceAccount from './helpers/device-account'
+import * as token from './helpers/token'
+import * as usedRefreshToken from './helpers/used-refresh-token'
+
+/**
+ * This class' purpose is to implement the interface needed by the OAuthProvider
+ * to interact with the account database (through the {@link AccountManager}).
+ *
+ * @note The use of this class assumes that there is no entryway.
+ */
+export class OAuthStore
```

Comment:
Most of this code was just moved from `./account-manager`. What changes is only the stuff related to the `AccountStore`:
- account creation
- binding of device-account
- reset password
- handle availability

See the diff of the `AccountStore` interface to be able to tell which method have changed.